### PR TITLE
Proposed 5.9 branch

### PR DESCRIPTION
**Pull request by unquietwiki**
The README has been updated to explain the changes, and the current state of the work. This does compile in the latest VS2017, and it successfully runs on Windows 10 x64 with my home networked Epson MFP. I also marked the code with a few remaining TODOs; alongside figuring out how to get rid of the remaining Roslynator security warnings.

### CODE REVIEW

**Comment by cyanfish** [comment]
For .NET 4.7.2, I don't want to change the .NET version since it would mean dropping support for older versions of Windows.
For CodeMaid/VS, you should be able to configure it based on the project conventions (variable case, namespace order, default namespaces) to minimize changes. I use Resharper for formatting.
For Roslynator, at least some of that stuff looks good, but I can't review 1700 files.

**Comment by unquietwiki** [comment]
Those are fair arguments. Regarding OS support, Vista & older are EOL; and
the Framework is upgradable on 7 & later. Do you have a cutoff on the old
stuff?

**Comment by cyanfish** [comment]
Right now NAPS2 supports as old as XP (and I know of people still using it there, despite EOL). Unless there's a compelling reason to migrate I'd rather maintain compatibility.

**Comment by unquietwiki** [comment]
* https://docs.microsoft.com/en-us/dotnet/framework/install/on-windows-xp
*
https://blogs.msdn.microsoft.com/dotnet/2017/10/17/announcing-the-net-framework-4-7-1/
* https://stackoverflow.com/questions/39050911/net-standard-net-core-pcl

MS only supports .NET 4 on XP; I believe you're already past that. And both
are EOL. MS is also replacing how they distribute & compile .NET apps: I've
already been able to do C# code that works on Linux & Windows easily. When
I was getting the assemblies to update, the newer way of importing those
wasn't locked to specific DLL versions. And it's hard to fix stuff in 10;
if stuff never got fixed between .NET versions since. I hope that helps.

**Comment by cyanfish** [comment]
Thanks. I think I'll stick with 4.0 for now.

**Comment by unquietwiki** [comment]
Well, how can we work on this going forward? I'd like to be able to have
something to hand off to you when you are ready. I don't have the bandwidth
or anti-pathy to fork this as NAPS3; but I don't want to abandon the work
either.

File: NAPS2.Core/Automation/AutomatedScanning.cs [hunk 1]
```diff
@@ -421,7 +419,7 @@ private bool IsPdfFile(string path)
         {
             string extension = Path.GetExtension(path);
             Debug.Assert(extension != null);
-            return extension.ToLower() == ".pdf";
+            return string.Equals(extension, ".pdf", StringComparison.CurrentCultureIgnoreCase);
```
Comment (timvdh) [inline NAPS2.Core/Automation/AutomatedScanning.cs]: What's in it?

**Comment by unquietwiki** [comment]
This was basically an attempt to refactor on newer .NET libraries, to make
it easier to debug & avoid bugs that have most likely been fixed in the 10+
years since Vista came out. Ideally, some consideration should be given to
Linux & Mac folks too, if the GUI stuff could be abstracted.

File: NAPS2.Core/Automation/AutomatedScanning.cs (hunk 1 above)
Comment (unquietwiki) [reply NAPS2.Core/Automation/AutomatedScanning.cs]: @timvdh Did you have a chance to see my other reply, or to try out a build of this? It's basically code refactoring against newer .NET Framework.

**Comment by unquietwiki** [comment]
@cyanfish @timvdh I'll close out this PR, since it looks like there's a 6.0 in the works now, and you two had some concerns regarding this. I'll figure out how to re-do this in a more acceptable manner, as able. Thanks!

**Pull request closed without merging**